hazelcast/src/main/java/com/hazelcast/query/impl/DefaultValueCollector.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.query.impl;
+
+import com.hazelcast.query.extractor.ValueCollector;
+import com.hazelcast.query.impl.getters.MultiResult;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DefaultValueCollector extends ValueCollector {
+
+    private List<Object> values = new ArrayList<Object>();
```

Comment:
this seems to be quite wasteful. I can imagine a lot of extractor will produce just a single value.